sdk/python/feast/infra/materialization/snowflake_engine.py

```diff
@@ -271,32 +271,63 @@ def _materialize_one(
 
             fv_latest_values_sql = offline_job.to_sql()
 
+            # Determine payload size if writing to an external online store
+            if self.repo_config.online_store.type != "snowflake.online":
+                if feature_view.entity_columns:
+                    join_keys = [entity.name for entity in feature_view.entity_columns]
+                    unique_entities = '"' + '", "'.join(join_keys) + '"'
+
+                    query = f"""
+                        SELECT
+                            {unique_entities}
```

Comment:
can you just do a count distinct in the query instead of getting a length of the individual rows